core/domain/story_domain.py

```diff
@@ -0,0 +1,359 @@
+from core.platform import models
+import feconf
+
+(story_models,) = models.Registry.import_models([models.NAMES.story])
+
+
+# Do not modify the values of these constants. This is to preserve backwards
+# compatibility with previous change dicts.
+STORY_PROPERTY_TITLE = 'title'
+STORY_PROPERTY_DESCRIPTION = 'description'
+STORY_PROPERTY_TOPIC = 'topic'
+STORY_PROPERTY_NOTES = 'notes'
+STORY_PROPERTY_LANGUAGE_CODE = 'language_code'
+STORY_PROPERTY_STORY_CONTENTS = 'story_nodes'
+
+# This takes additional 'property_name' and 'new_value' parameters and,
+# optionally, 'old_value'.
+CMD_UPDATE_STORY_PROPERTY = 'update_story_property'
+
+
+class StoryChange(object):
+    """Domain object for changes made to story object."""
+    STORY_PROPERTIES = (
+        STORY_PROPERTY_TITLE, STORY_PROPERTY_DESCRIPTION,
+        STORY_PROPERTY_TOPIC, STORY_PROPERTY_NOTES,
+        STORY_PROPERTY_STORY_CONTENTS, STORY_PROPERTY_LANGUAGE_CODE)
+
+    def __init__(self, change_dict):
+        """Initialize a StoryChange object from a dict.
+
+        Args:
+            change_dict: dict. Represents a command. It should have a 'cmd'
+                key, and one or more other keys. The keys depend on what the
+                value for 'cmd' is. The possible values for 'cmd' are listed
+                below, together with the other keys in the dict:
+                - 'update story property' (with property_name, new_value
+                and old_value)
+
+        Raises:
+            Exception: The given change dict is not valid.
+        """
+        if 'cmd' not in change_dict:
+            raise Exception('Invalid change_dict: %s' % change_dict)
+        self.cmd = change_dict['cmd']
+
+        if self.cmd == CMD_UPDATE_STORY_PROPERTY:
+            if (change_dict['property_name'] in
+                    self.STORY_PROPERTIES):
+                self.property_name = change_dict['property_name']
+                self.new_value = change_dict['new_value']
+                self.old_value = change_dict.get('old_value')
+            else:
+                raise Exception('Invalid change_dict: %s' % change_dict)
+
+    def to_dict(self):
+        """Returns a dict representing StoryChange domain object.
+
+        Returns:
+            dict. A dict representing StoryChange instance.
+        """
+        return {
+            'cmd': self.cmd,
+            'property_name': self.property_name,
+            'new_value': self.new_value,
+            'old_value': self.old_value
+        }
+
+
+class StoryNode(object):
+    """Domain object describing a node in the exploration graph of a
+    story.
+    """
+
+    def __init__(
+            self, node_id, destination_nodes,
+            acquired_skill_ids, prerequisite_skill_ids,
+            annotations, exploration_id):
+        """Initializes a StoryNode domain object.
+
+        Args:
+            node_id: str. The unique id for each node.
+            destination_nodes: list(str). The list of destination node ids that
```

Comment:
If this is a list of destination node IDs, call it destination_node_ids and not destination_nodes.

Reply:
Done.